### PR TITLE
[Opt][Vectorized] Merge block in scanner to speed up query with conjunct

### LINKED ISSUE #8394: [Enhancement] Merge block rows lower than batch size to speed up query
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

After we do segment vectorized, some qurey like ssb q1 have conjunct will cause the block lower than batch size, so we should merge block to speed up query.

Before
SSB Q1: 0.7s

After
SSB Q1:0.6s

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #8394 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/volap_scan_node.cpp [hunk 1]
```diff
@@ -221,7 +221,14 @@ void VOlapScanNode::scanner_thread(VOlapScanner* scanner) {
             std::lock_guard<std::mutex> l(_free_blocks_lock);
             _free_blocks.emplace_back(block);
         } else {
-            blocks.push_back(block);
+            if (!blocks.empty() && blocks.back()->rows() + block->rows() <= _runtime_state->batch_size()) {
```
Comment (morningman) [inline be/src/vec/exec/volap_scan_node.cpp:224]: Do we need a threshold? For example, if the full block size is 1024, but each coming block size is 1023, than we have to  merge block everytime, and there is even a `mutex` here

Comment (HappenLee) [reply be/src/vec/exec/volap_scan_node.cpp:224]: `blocks.back()->rows() + block->rows() <= _runtime_state->batch_size() `
1023 + 1023 > 1024
so do not do merge here

**Review by morningman — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**